File: leetcode/Single Number III/1.kt
class Solution {
    fun singleNumber(nums: IntArray): IntArray {
        nums.sort()
        val result = mutableListOf<Int>()
        for (i in 0 until nums.size) {
            val leftSame = if (i - 1 >= 0) {
                nums[i] != nums[i - 1]
            } else true
            val rightSame = if (i + 1 < nums.size) {
                nums[i] != nums[i + 1]
            } else true

            if (leftSame && rightSame) {
                result.add(nums[i])
            }
        }
        return result.toIntArray()
    }
}

File: leetcode/Single Number III/1.go
func singleNumber(nums []int) []int {
	m := make(map[int]int)
	for _, num := range nums {
		m[num]++
	}

	result := []int{}
	for num, count := range m {
		if count == 1 {
			result = append(result, num)
		}
		if len(result) == 2 {
			break
		}
	}
	return result
}